lisa/analysis/rta.py

```diff
@@ -0,0 +1,339 @@
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from numpy import unique
+
+from lisa.utils import memoized
+from lisa.analysis.base import AnalysisHelpers
+from lisa.analysis.base import TraceAnalysisBase
+from lisa.trace import requires_events
+
+class Events(TraceAnalysisBase):
+    """
+    Support for RTA events analysis.
+
+    :param trace: input Trace object
+    :type trace: :class:`trace.Trace`
+    """
+
+    name = 'rta'
+
+    def __init__(self, trace):
+        """
+        Load peformance data of an rt-app workload
+        """
+        logger = self.get_logger()
+
+        self.trace = trace
+
+        if 'rtapp_task' not in self.trace.base_trace.available_events:
+            raise ValueError('No rt-app tasks events found in trace')
+
+        tasks_df = self.trace.base_trace.df_events('rtapp_task')
+        self.rtapp_tasks = unique(tasks_df["__comm"].values).tolist()
+
+        logger.info("Found rt-app tasks: %s", self.rtapp_tasks)
+
+    def _task_filtered(self, df, task=None):
+        if not task:
+            return df
+        if task not in self.rtapp_tasks:
+            raise ValueError("Task [{}] is not an rt-app task: {}"
+                             .format(task, self.rtapp_tasks))
+        return df[df["__comm"] == task]
+
+
+###############################################################################
+# DataFrame Getter Methods
+###############################################################################
+
+    ###########################################################################
+    # rtapp_main events related methods
+    ###########################################################################
+
+    @requires_events('rtapp_main')
+    def df_rtapp_main(self):
+        """
+        Dataframe of events generated by the rt-app main task.
+
+        :returns: a :class:`pandas.DataFrame` with:
+
+          * A ``__comm`` column (the actual rt-app trace task name)
+          * A ``__cpu``  column (the CPU on which the task was running at event
+                                generation time)
+          * A ``__line`` column (the ftrace line numer)
+          * A ``__pid``  column (the PID of the task)
+          * A ``data``   column (the data corresponding to the reported event)
+          * An ``event`` column (the event generated)
+
+        TBD: define what are the valida values for ``data`` and ``event``
+        """
+        return self.trace.base_trace.df_events('rtapp_main')
+
+    @property
+    @df_rtapp_main.used_events
+    def rtapp_window(self):
+        df = self.df_rtapp_main()
+        return (
+            df[df.event == 'start'].index.values[0],
```

Comment:
filtering should be split in a variable so it's only done once

Reply:
Not sure what you mean here?!?

Reply:
`df[df.event == 'start']` goes through the whole series to create a series of bool, and then uses that series of bool to filter the original series, so doing it once when possible is a better idea:
```
        df = self.df_rtapp_main()
        index  = df[df.event == 'start'].index
        return (
            index.values[0],
            whatever_github_felt_like_annoyingly_cropping
       )
```